plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,342 @@
+version_added: '5.6.0'
+
+options:
+  src_iso:
+    description:
+    - This is the path of source ISO file.
+    - Will fail if specified ISO file does not exist on local machine.
+    type: str
+    required: true
+  dest_iso:
+    description:
+    - The path with file name of the customized ISO file on local machine.
+    type: str
+    required: true
+  delete_files:
+    description:
+    - Absolute paths for files inside the ISO file that should be removed.
+    type: list
+    required: false
+    elements: str
+  add_files:
+    description:
+    - Allows to add and replace files in the ISO file.
+    - Will create intermediate folders inside the ISO file when they do not exist.
+    type: list
+    required: false
+    elements: dict
+    suboptions:
+      src_file:
+        description:
+        - The path with file name on the machine the module is executed on.
+        type: path
+      dest_file:
+        description:
+        - The absolute path of the file inside the ISO file.
+        type: str
+'''
+
+EXAMPLES = r'''
+- name: "Customize ISO file"
+  community.general.iso_customize:
+    src_iso: "/path/to/ubuntu-22.04-desktop-amd64.iso"
+    dest_iso: "/path/to/ubuntu-22.04-desktop-amd64-customized.iso"
+    delete_files:
+      - "/preseed/ubuntu.seed"
+    add_files:
+      - src_file: "/path/to/grub.cfg"
+        dest_file: "/boot/grub/grub.cfg"
+      - src_file: "/path/to/ubuntu.seed"
+        dest_file: "/preseed/ubuntu.seed"
+  register: customize_iso_result
+'''
+
+RETURN = r'''
+src_iso:
+  description: Path of source ISO file.
+  returned: on success
+  type: str
+  sample: "/path/to/file.iso"
+dest_iso:
+  description: path of the customized iso file.
+  returned: on success
+  type: str
+  sample: "/path/to/customized.iso"
+'''
+
+import os
+import traceback
+import copy
+
+PYCDLIB_IMP_ERR = None
+try:
+    import pycdlib
+    HAS_PYCDLIB = True
+except ImportError:
+    PYCDLIB_IMP_ERR = traceback.format_exc()
+    HAS_PYCDLIB = False
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.common.text.converters import to_native
+
+MODULE_ISO_CUSTOMIZE = None
+ISO_MODE = "iso9660"
+
+
+# The upper dir exist, we only add subdirectoy
+def iso_add_dir(opened_iso, dir_path):
+    parent_dir, check_dirname = dir_path.rsplit("/", 1)
+    if not parent_dir.strip():
+        parent_dir = "/"
+    check_dirname = check_dirname.strip()
+
+    for dirname, dirlist, dummy_filelist in opened_iso.walk(iso_path=parent_dir.upper()):
+        if dirname == parent_dir.upper():
+            if check_dirname.upper() in dirlist:
+                return 0, ""
+
+            if parent_dir == "/":
+                current_dirpath = "/%s" % check_dirname
+            else:
+                current_dirpath = "%s/%s" % (parent_dir, check_dirname)
+
+            try:
+                if ISO_MODE == "iso9660":
+                    opened_iso.add_directory(current_dirpath.upper())
```

Comment:
maybe add:
```python
current_dirpath = current_dirpath.upper()
```
after line 125, then simplify the calls on lines 129, 131, 133 and 135?

Reply:
Fix it. Thanks